src/main/java/cz/student/richard808code/UserDao.java

```diff
@@ -0,0 +1,162 @@
+package cz.student.richard808code;
+
+import cz.common.DbUtil;
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+public class UserDao {
+
+    private static final String CREATE_USER_QUERY =
+            "INSERT INTO users(username, email, password) VALUES (?, ?, ?)";
+
+    private static final String READ_USER_QUERY =
+            "SELECT * FROM users WHERE id = ?";
+
+    private static final String UPDATE_USER_QUERY =
+            "UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?";
+
+    private static final String DELETE_USER_QUERY =
+            "DELETE FROM users WHERE id = ?";
+
+    private static final String SELECT_ALL_USERS_QUERY =
+            "SELECT * FROM users";
+
+    public User create(User user) {
+        try (Connection conn = DbUtil.getConnection()) {
+
+            PreparedStatement preStmt = conn.prepareStatement(CREATE_USER_QUERY, PreparedStatement.RETURN_GENERATED_KEYS);
+            preStmt.setString(1, user.getUserName());
+            preStmt.setString(2, user.getEmail());
+            preStmt.setString(3, hashPassword(user.getPassword()));
+            preStmt.executeUpdate();
+
+            ResultSet rs = preStmt.getGeneratedKeys();
+            if (rs.next()) {
+                int id = rs.getInt(1);
+                System.out.println("Inserted ID: " + id);
+                user.setId(id);
+            }
+
+        } catch (SQLException e) {
+            System.err.println("ERROR: Cannot connect to database");
```

Comment:
CLEAN-CODE: nemusi se jednat vzdy jenom o problem pripojeni do databaze, ale treba tabulka user nemusi existovat, dal bych obecnejsi popis chyby.

```suggestion
System.err.println("Problem with create user in database: " + e.getMessage());
```

Reply:
Opraveno